lib/merge.ts

```diff
@@ -374,9 +374,28 @@ export function normalizeSourceFile(f: ts.SourceFile, fc: FacadeConverter) {
                       if (base.ident(member.name) === 'prototype') {
                         break;
                       }
-                      addModifier(member, ts.createNode(ts.SyntaxKind.StaticKeyword));
-                      member.parent = existing;
-                      Array.prototype.push.call(members, member);
+
+                      // Finds all existing declarations of this property in the inheritance
+                      // hierarchy of this class
+                      const existingDeclarations =
+                          findPropertyInHierarchy(base.ident(member.name), existing, classes);
+
+                      if (existingDeclarations.size) {
+                        // TODO(derekx): For dom.d.ts it makes sense to make all properties that are
+                        // declared on the anonymous types of top level variable declarations
+                        // static, but this may not always be correct
+                        for (const existingDecl of existingDeclarations) {
+                          addModifier(existingDecl, ts.createModifier(ts.SyntaxKind.StaticKeyword));
+                        }
+                      }
+
+                      // If needed, add declaration of property to the interface that we are
+                      // currently handling
+                      if (!findPropertyInClass(base.ident(member.name), existing)) {
+                        addModifier(member, ts.createModifier(ts.SyntaxKind.StaticKeyword));
+                        member.parent = existing;
```

Comment:
don't modify the TS ast as part of running the emitter. Instead keep whatever information you need in separate maps on the side.
Otherwise we could get really strange bugs.